### ISSUE TITLE
feat: add no-empty-links rule (MD042)

### ISSUE THREAD

**Issue by shinagawa-web** [shinagawa-web/gomarklint#106]
## Overview

Implement the `no-empty-links` rule (equivalent to markdownlint MD042 / remark-lint `no-empty-url`).

Closes part of #76.

## Rule Description

Links must not have an empty URL destination. Empty links are likely placeholders or mistakes that should be resolved before publishing.

**Invalid:**
```markdown
[link text]()
[link text](#)
[link text](<>)
```

**Valid:**
```markdown
[link text](https://example.com)
[link text](#section)
[link text](./page.md)
```

## Implementation Notes

- Rule key: `no-empty-links`
- Detect Markdown links where the URL destination is empty (`()`), contains only a fragment placeholder (`(#)`), or is blank inside angle brackets (`(<>)`)
- Links inside fenced code blocks and inline code spans should be ignored
- Images (`![alt]()`) should also be checked
- Config field: boolean toggle (default enabled)
